packages/pigeon/lib/kotlin_generator.dart

```diff
@@ -828,6 +864,14 @@ String _cast(String variable, {required TypeDeclaration type}) {
   if (typeString == 'Int' || typeString == 'Long') {
     return _castInt(type.isNullable);
   }
+  if (isEnum(root, type)) {
+    if (type.isNullable) {
+      return '(list[13] as Int?)?.let {\n'
```

Comment:
I suspect this `list[13]` isn't going to work for all input files ;)

Reply:
that's just a restriction of the new feature. I'll add documentation...